### PR TITLE
feat: create Dialog manager

### PR DESCRIPTION
**Pull request by adonyssantos**

### CODE REVIEW

File: src/hooks/useDialog.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { atom, useAtom } from 'jotai'
+
+const dialogAtom = atom(false)
+
+export function useDialog(){
+  const [isOpen, setIsOpen] = useAtom(dialogAtom)
```
Comment (Alan-TheGentleman) [inline src/hooks/useDialog.ts:6]: Use setIsOpen directly, no need for other methods

File: src/App.tsx [hunk 2]
```diff
@@ -1,8 +1,21 @@
+import { Dialog } from './components/Dialog/Dialog'
+import { useDialog } from './hooks/useDialog'
 import TaskListPage from './pages/task-list/TaskListPage'
-import './styles/reset.css'
+// import './styles/reset.css'
```
Comment (Alan-TheGentleman) [inline src/App.tsx]: remove the line

File: src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import type { ReactNode } from "react";
+
+export interface DialogProps extends Omit<MuiDialogProps, 'open' | 'onClose'> {
+  children: ReactNode;
+};
+
+export function Dialog({ children, ...rest }: DialogProps) {
+  const { isOpen, closeDialog } = useDialog();
+
+  return <StyledDialog open={isOpen} onClose={closeDialog} {...rest}>{children}</StyledDialog>;
```
Comment (Alan-TheGentleman) [inline src/components/Dialog/Dialog.tsx]: Removed the styled if not used

**Review by Alan-TheGentleman — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 2 above)
Comment (adonyssantos) [reply src/App.tsx]: css reset will be removed because it affects the styles of mui

File: src/hooks/useDialog.ts (hunk 1 above)
Comment (adonyssantos) [reply src/hooks/useDialog.ts:6]: got it

File: src/styles/reset.css
Comment (adonyssantos) [inline src/styles/reset.css:1]: css reset will be removed because it affects the styles of mui

File: src/components/Dialog/Dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { StyledDialog } from "./styles";
+import { useDialog } from "../../hooks/useDialog";
+import type { DialogProps as MuiDialogProps } from "@mui/material";
+import type { ReactNode } from "react";
+
+export interface DialogProps extends Omit<MuiDialogProps, 'open' | 'onClose'> {
```
Comment (Alan-TheGentleman) [inline src/components/Dialog/Dialog.tsx:6]: Great !

**Review by Alan-TheGentleman — APPROVED** [review]
LGTM !

**Comment by adonyssantos** [comment]
closed because a different implementation was created in PR #9

**Pull request closed without merging**